src/node_api.cc

```diff
@@ -557,65 +576,67 @@ void napi_module_register(napi_module* mod) {
   node::node_module_register(nm);
 }
 
-#define RETURN_STATUS_IF_FALSE(condition, status)                       \
+#define RETURN_STATUS_IF_FALSE(env, condition, status)                  \
   do {                                                                  \
     if (!(condition)) {                                                 \
-      return napi_set_last_error((status));                             \
+      return napi_set_last_error((env), (status));                      \
     }                                                                   \
   } while (0)
 
-#define CHECK_ARG(arg) RETURN_STATUS_IF_FALSE((arg), napi_invalid_arg)
+#define CHECK_ENV(env) \
+  assert((env) != nullptr)
```

Comment:
That particular call to the NAPI function can't continue with a null `env`. But that doesn't mean other calls to other NAPI functions in the same callback or other callbacks in the same native module, or other modules can't continue to work.

Why should this invalid argument error be treated any differently from any other invalid argument error? An invalid argument almost always indicates a bug in the code, but aborting the process at any sign of a bug _in an external module_ seems extreme to me.